### DISCUSSION TITLE
Nous A1: Built in RGB LED control possible?

### DISCUSSION THREAD

**Discussion by hagenbuch** [arendst/Tasmota#16246] [Q&A]
Dear all,

I managed to flash Tasmota 12.0.2 on four Nous A1 plugs.

As far as I read, they might contain a RGB LED right on the PCB.

Can I control them and how? 

I've read the color options but am not sure if they refer to the internel LED and how...

**Comment by barbudor** [comment]
I have a few of them and they don't have such a RGB led

Now, it may be possible that they have added that on recent model.
I believe you did opened them for flashing so would you mind posting a good photo of the PCB?

**Comment by hagenbuch** [comment]
The PCB looks pretty much like this 

https://user-images.githubusercontent.com/80213345/144631046-d0e11031-9cb8-42bc-8aee-edc8d531f4d7.jpg

Twi LEDs: 3LED1 and 3LED2. Now if we knew what "3" means :)

I would speculate that 3 wires could well be possible.. 

I guess that the template should know about RGB LEDs,

- GPIO 0 is set to "Led_i" and "1"
- GPIO 2 is set to "Led_Link_i"

The other GPIOs deal with the relay, the button and supposedly with the chip measuring the power.

I also didn't find any video mentioning color lights on the plug but if you look into the socket, they printed a rainbow colored circle segment. Would be nice if you could see red means over 2000 watt, yellow over 500, green over 20 and blue below or something.

**Reply by barbudor** [reply]
I have that exact model you gave the photos and the leds are basic single color, 2 poles (Anode Cathode).

There are some plugs with a similar casing which are said to havea RGB leds but AFAIK, those are not Tasmota compatible

If you are a hacker, it is still possible to remove one of the origina led and try to fit a WS2812 in 3030 package

**Reply by sfromis** [reply]
Since nearly every component label on the board is prefixed with "3", this does not seem to tell anything about the components. And that "rainbow" does not appear to be anything more than to look "cool", without functional implications. Potentially to match some color scheme used in their app.

**Reply by TheSmartGerman** [reply]
Well RGB should be no problem: https://tasmota.github.io/docs/Lights/#addressable-leds

over here are better pictures: https://templates.blakadder.com/nous_A1.html

I just order a set auf ws2812 in **2020** size (2x2mm) - should fit into the space.

https://www.aliexpress.com/i/2251832850516474.html?gatewayAdapt=deu2glo4itemAdapt

**Reply by sfromis** [reply]
> Well RGB should be no problem

That's only if you do not consider it a "problem" to replace a two-pin led by a 4-pin neopixel, needing additional wiring. Not impossible with some careful work, but not a quick and easy task, unless your soldering skills are pretty good.

Good luck! You may need it :stuck_out_tongue_closed_eyes:

**Comment by hagenbuch** [comment]
Well I guess I'll have to agree :)

Thanks for all your responses!
